File: assignments/roman_to_integer/RomanToIntegerTestRunner.java
public class RomanToIntegerTestRunner {

    public static void main(String[] args) {
        RomanToInteger converter = new RomanToInteger();

        String[] inputs = {"III", "LVIII", "MCMXCIV", "IX", "IV", "CM"};
        int[] expectedOutputs = {3, 58, 1994, 9, 4, 900};

        int passedCount = 0;
        StringBuilder report = new StringBuilder();

        System.out.println("===== Starting Automated Tests for RomanToInteger =====");

        for (int i = 0; i < inputs.length; i++) {
            try {
                int actualOutput = converter.convert(inputs[i]);

                report.append("Test Case #").append(i + 1).append(": Input '").append(inputs[i]).append("'\n");
                report.append(" -> Expected: ").append(expectedOutputs[i]).append(", Got: ").append(actualOutput);

                if (actualOutput == expectedOutputs[i]) {
                    report.append(" -> ✅ PASS\n\n");
                    passedCount++;
                } else {
                    report.append(" -> ❌ FAIL\n\n");
                }
            } catch (Exception e) {
                report.append("Test Case #").append(i + 1).append(": Input '").append(inputs[i]).append("'\n");
                report.append(" -> ❌ ERROR: ").append(e.getMessage()).append("\n\n");
            }
        }

        System.out.println(report.toString());
        System.out.println("===== Test Summary =====");
        System.out.println("Total Tests: " + inputs.length);
        System.out.println("Passed: " + passedCount);
        System.out.println("Failed: " + (inputs.length - passedCount));
        System.out.println("========================");

        System.out.println("[SUMMARY]:" + passedCount + ":" + inputs.length);
    }
}

File: assignments/roman_to_integer/python/roman_to_integer_Test_runner.py
import sys
sys.path.append('.')
from roman_to_integer import RomanToInteger

def run_tests():
    solver = RomanToInteger()

    test_cases = [
        ("III", 3),
        ("LVIII", 58),
        ("MCMXCIV", 1994),
        ("IX", 9),
        ("IV", 4),
        ("CM", 900),
    ]

    passed_count = 0
    report = []

    print("===== Starting Automated Tests for RomanToInteger (Python) =====")

    for i, (input_val, expected_output) in enumerate(test_cases):
        try:
            actual_output = solver.convert(input_val)

            report.append(f"Test Case #{i + 1}: Input '{input_val}'")
            report.append(f" -> Expected: {expected_output}, Got: {actual_output}")

            if actual_output == expected_output:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(f"Test Case #{i + 1}: Input '{input_val}'")
            report.append(f" -> ❌ ERROR: {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary =====")
    print(f"Total Tests: {len(test_cases)}")
    print(f"Passed: {passed_count}")
    print(f"Failed: {len(test_cases) - passed_count}")
    print("========================")

    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    run_tests()